coral-erm/coral
# Export Improvements

## This PR provides two major improvements:

1. It speeds up the resources module by doing fewer joins and trying to only do joins that are necessary (which also helps with the export).
2. It supports exporting notes (as a json array)

## Requirements:

1. Run `resources/install/protected/next/001-16.sql ` against your development database. I'm not sure what needs to happen process-wise for this sql to be part of the next update cycle but that needs to happen...

## Things to test:

1. Go to `Admin` -> `Export Configuration` and try changing the settings navigating away and ensuring that they save.
2. With notes enabled, run an export and check that notes are included (probably the last column of the CSV). Check that the right notes appear.
3. Check that parent, child, and notes columns appear if and only if they are enabled in the export configuration options.
4. Under `Home`, try each of the search options. **Note:** It is worth checking that all these options still work correctly because this PR modifies joins and so while developing it, I definitely broke the search a couple of times.
